### PR TITLE
Admin test: Use settings instead of mocker

### LINKED ISSUE #2316: Chore: use `settings` fixture in unit test `test_pre_login_user_add_staff_to_group`
> Ah good catch. I think this means `test_pre_login_user_add_staff_to_group`, which existed before this PR, should do the same

_Originally posted by @angela-tran in https://github.com/cal-itp/benefits/pull/2315#discussion_r1725950151_

### PR DESCRIPTION
**Pull request by machikoyasuda**
fixes #2316

### CODE REVIEW

**Review by angela-tran — DISMISSED** [review]
👍

File: tests/pytest/core/test_admin.py [hunk 1]
```diff
@@ -71,8 +71,7 @@ def test_pre_login_user_add_staff_to_group(mocker, model_AdminUser):
     mocked_request = mocker.Mock()
     mocked_request.session.get.return_value = None
 
-    mocker.patch.object(benefits.core.admin.settings, "GOOGLE_SSO_STAFF_LIST", [model_AdminUser.email])
-
+    settings.GOOGLE_SSO_STAFF_LIST = [model_AdminUser.email]
```
Comment (thekaveman) [inline tests/pytest/core/test_admin.py:74]: I don't think this is quite right. This is the `settings` object imported from `django.conf`

It should be the `settings` fixture from `pytest-django` (e.g. injected to the test method).

See https://github.com/cal-itp/benefits/blob/main/tests/pytest/test_secrets.py#L80

**Review by thekaveman — CHANGES_REQUESTED** [review]

File: tests/pytest/core/test_admin.py (hunk 1 above)
Comment (machikoyasuda) [reply tests/pytest/core/test_admin.py:73]: https://github.com/cal-itp/benefits/pull/2347/commits/d35edf76ad79dfa2d958f265da998e1cd7368954

**Review by thekaveman — APPROVED** [review]
Nice :+1:

**Pull request merged**